src/api/semesterOperations.tsx

```diff
@@ -0,0 +1,41 @@
+import { Semester } from 'api';
+import { AddSemesterData } from 'components/AddSemesterModal';
+
+const handleErrors = (response: Response) => {
+	if (!response.ok) {
+		throw new Error('fetch unsuccessful');
+	}
+	return response;
+};
+
+// create
+export const fetchPostSemester = async (data: AddSemesterData) =>
```

Comment:
I think we should call this `createSemester` and we can name the following `getSemester`, `updateSemester` and `deleteSemester`. I think these names are the most intuitive but if you have a better idea definitely use it.